plugins/inputs/metric_streams_listener/metric_streams_listener.go

```diff
@@ -0,0 +1,411 @@
+package metric_streams_listener
+
+import (
+	"compress/gzip"
+	"crypto/tls"
+	_ "embed"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"math"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/choice"
+	tlsint "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/selfstat"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+// defaultMaxBodySize is the default maximum request body size, in bytes.
+// if the request body is over this size, we will return an HTTP 413 error.
+// 500 MB
+const defaultMaxBodySize = 500 * 1024 * 1024
+
+type MetricStreamsListener struct {
```

Comment:
I know you've already renamed this once, but I'm wondering if this is the best name for the plugin. Metric streams is part of cloudwatch, and we have a cloudwatch input plugin already. I'm not an AWS expert, so I'm not sure how much overlap there is between this new plugin and telegraf's existing cloudwatch plugin. Would it make sense to call this "cloudwatch metric streams" or "cloudwatch listener"?

Does the cloudwatch input handle the same aws metrics as this new plugin? My impression from quickly scanning the aws docs is that metric streams is a push model and cloudwatch is pull but they do handle the same source metrics.

If that's the case, I think this should have a similar name to the existing cloudwatch plugin so people get the impression that they do similar jobs.

Reply:
I'm open to renaming it sure, is "cloudwatch_metric_streams_listener" too long?

Reply:
"cloudwatch_metric_streams_listener" is a mouthful. AWS's product branding word salad doesn't make it easy.

Many similar telegraf input plugins do have suffixes like "_listener" or "_consumer". There's even one case of "_push" with GCP's pubsub. The suffixes are used to disambiguate. For example "http" is pull, "http_listener" is push. "kafka" gathers metrics about the kafka server's performance and "kafka_consumer" receives kafka messages and parses them as metric data.

My vote is "cloudwatch_metric_streams" with no "_listener". It has a few things going for it:
* It's AWS's official product name
* It's similar to the existing "cloudwatch" plugin and will show up next to it in an alphabetical list.
* It's already not ambiguous. To use it, you have to be familiar with the AWS service. If you're familiar with the service you'll know "cloudwatch" is pull and "cloudwatch_metric_streams" is push.